### PR TITLE
feat(lint): fix global-require eslint rule

### PR DESCRIPTION
**Pull request by mvorisek**

### CODE REVIEW

**Comment by mvorisek** [comment]
PR is complete and it should not contain any breaking change, so this can go into the develop branch without having to wait for branch cut for 2.10.x.

Node 12 support is kept, Node 14+ is however required for eslint as described in the PR description.

File: src/definitions/modules/sidebar.js [hunk 1]
```diff
@@ -309,7 +309,7 @@
                     module.verbose('Forcing repaint event');
                     element.style.display = 'none';
                     var ignored = element.offsetHeight;
-                    element.scrollTop = element.scrollTop;
+                    element.scrollTop = element.scrollTop; // eslint-disable-line no-self-assign
```
Comment (mvorisek) [inline src/definitions/modules/sidebar.js:312]: is this line really useful? can we unify the (forced) repaint across modules?

File: src/definitions/modules/calendar.js [hunk 2]
```diff
@@ -758,13 +777,13 @@
                         var ms1d = 864e5, // milliseconds in a day
                             ms7d = 7 * ms1d; // milliseconds in a week
 
-                        return function () { // return a closure so constants get calculated only once
+                        return (function () { // return a closure so constants get calculated only once
```
Comment (mvorisek) [inline src/definitions/modules/calendar.js:780]: the closure is invoked immediately and not returned, is it really useful?

File: src/definitions/modules/dropdown.js [hunk 3]
```diff
@@ -2391,7 +2397,7 @@
                         $nextSelectedItem
                             .addClass(className.selected)
                         ;
-                        if (settings.selectOnKeydown && module.is.single() && !$nextItem.hasClass(className.actionable)) {
+                        if (settings.selectOnKeydown && module.is.single() && !$nextSelectedItem.hasClass(className.actionable)) {
```
Comment (mvorisek) [inline src/definitions/modules/dropdown.js:2400]: did I fixed the name to an intended variable?

File: src/definitions/behaviors/api.js [hunk 4]
```diff
@@ -157,11 +157,6 @@
                 },
                 write: {
                     cachedResponse: function (url, response) {
-                        if (response && response === '') {
```
Comment (mvorisek) [inline src/definitions/behaviors/api.js:160]: never true condition, not sure what to do

if response can be a plain text, caching empty response might be expected, as the condition is never true, no BC break

**Review by lubber-de — CHANGES_REQUESTED** [review]
I havent reviewed all, but those 2 changes i wanted to report beforehand

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -16,7 +16,6 @@ jobs:
         ref: master
     - uses: actions/setup-node@v3
       with:
-        node-version: 14
```
[the remark this answers is not in the document]
Comment (mvorisek) [reply .github/workflows/release.yml:19]: this is a CI script that use `ubuntu-latest` Docker image, so this way, it will use the latest stable Node which should be always supported by Fomantic-UI

[the remark this answers is not in the document]
Comment (lubber-de) [reply .github/workflows/release.yml:19]: The latest stable node does not necessarily mean that the nightly script will still work in any future version! If this isnt tested for node 22 (for example) and the docker images gets updated and, by whatever (unlikely) incompatibility the scripts does not work (or throws any unexpected error) it will break
Yes, unlikely it breaks, but who knows. As said, even setup-node suggests to state an explicit version 

Remember, we had the situation where Node 15/16 was previouly not working with fomantic anymore because of an incompatibility with inquirere not supporting yarn with interactive scripts . We had this before, that's why :wink:

Yes, Fomantic _should_ support the latest node version, but only until tested

[the remark this answers is not in the document]
Comment (mvorisek) [reply .github/workflows/release.yml:19]: I prefer no fixed numbers anywhere so I added `latest` to the CI.

- the release script (outside the push to registry) is the same as our CI test
- the nighly script is the same as well plus one file to compute the next version

so release script should never fail and nighly script can fail, but I would say nighly failure is good, as we can detect the problems :)

File: src/definitions/modules/sidebar.js (hunk 1 above)
Comment (lubber-de) [reply src/definitions/modules/sidebar.js:312]: > is this line really useful?

From the git history, is is not clear, why this line was necessary (clear commit message ftw). I strongly believe it was needed by IE8-11 (?) back then.

>  can we unify the (forced) repaint across modules?

Otherwise, there are a lot of functions i would remove from all modules and put into a separate "core.js" or similar  (like the debug functions). maybe enhance the existing site.js. 

Perhaps repaint is also a candidate.

But this would mean each module is not working on its own anymore (at least when following the ES5 syntax as it it now).
I am already thinking about such way to reduce the code overhead. Even if that means people have to make sure to include core.js...but this is another story

File: src/definitions/modules/calendar.js (hunk 2 above)
Comment (lubber-de) [reply src/definitions/modules/calendar.js:780]: > the closure is not returned (invoked immediately)

I think it is returned (wouldn't work totherwise)
```javascript
return function () { // return a closure so constants get calculated only once
...
}();
```
>  is it really useful?

tbh, i just took that calculation function from the credited website back then 😆

File: src/definitions/modules/dropdown.js (hunk 3 above)
Comment (lubber-de) [reply src/definitions/modules/dropdown.js:2400]: You actually found a bug and fixed it. 👍🏼 Deserves a separate PR, please.

File: src/definitions/behaviors/api.js (hunk 4 above)
Comment (lubber-de) [reply src/definitions/behaviors/api.js:160]: Agreed, lets remove this

File: tasks/config/npm/gulpfile.js [hunk 6]
```diff
@@ -2,33 +2,33 @@
  *           Set-up
  *******************************/
 
-var
+let
     gulp   = require('gulp'),
 
     // read user config to know what task to load
-    config = require('./tasks/config/user')
+    config = require('./tasks/config/user') // eslint-disable-line import/extensions
```
Comment (mvorisek) [inline tasks/config/npm/gulpfile.js:9]: this file is copied in two places using `gulp.src(source.userGulpFile)` - do we really need that instead of the `/gulpfile.js`

please notice `tasks/config/npm/gulpfile.js` vs. `/gulpfile.js` which are almost identical

File: src/definitions/modules/calendar.js (hunk 2 above)
Comment (mvorisek) [reply src/definitions/modules/calendar.js:780]: `let x = (function () {})( return 5; );` (function evaluated immediatelly) is basically a `let x = 5;`

maybe it has something to do with https://eslint.org/docs/latest/rules/wrap-iife

still, I think "function evaluated immediatelly" is not needed here

File: src/definitions/modules/dropdown.js (hunk 3 above)
Comment (mvorisek) [reply src/definitions/modules/dropdown.js:2400]: fixed in #2591

File: src/definitions/behaviors/api.js (hunk 4 above)
Comment (mvorisek) [reply src/definitions/behaviors/api.js:160]: fixed in #2591

File: tasks/admin/components/init.js [hunk 7]
```diff
@@ -38,6 +37,8 @@ const
 ;
 
 module.exports = function (callback) {
+    const github = require('../../config/admin/github'); // eslint-disable-line global-require
```
Comment (lubber-de) [inline tasks/admin/components/init.js:40]: Why was this necessary? If you leave the github const where it was before the eslint rule is already fulfilled, isn't it?
Now, you moved it here and have to disable the rule... i don't get the reason 🤔

File: tasks/admin/components/update.js [hunk 8]
```diff
@@ -36,6 +35,8 @@ const
 ;
 
 module.exports = function (callback) {
+    const github = require('../../config/admin/github'); // eslint-disable-line global-require
```
Comment (lubber-de) [inline tasks/admin/components/update.js:38]: same here. If you havent moved it, it was alreay a global const

File: tasks/admin/distributions/init.js [hunk 9]
```diff
@@ -38,6 +37,8 @@ const
 ;
 
 module.exports = function (callback) {
+    const github = require('../../config/admin/github'); // eslint-disable-line global-require
```
Comment (lubber-de) [inline tasks/admin/distributions/init.js:40]: and again here... 🙂

File: tasks/admin/distributions/update.js [hunk 10]
```diff
@@ -36,6 +35,8 @@ const
 ;
 
 module.exports = function (callback) {
+    const github = require('../../config/admin/github'); // eslint-disable-line global-require
```
Comment (lubber-de) [inline tasks/admin/distributions/update.js:38]: and again...

File: tasks/collections/build.js [hunk 11]
```diff
@@ -2,17 +2,15 @@
...
-        watch       = require('../watch'),
+const watch = require('../watch');
 
-        build       = require('../build'),
-        buildJS     = require('../build/javascript'),
-        buildCSS    = require('../build/css'),
-        buildAssets = require('../build/assets')
-    ;
+// build sub-tasks
+const build = require('../build');
```
Comment (lubber-de) [inline tasks/collections/build.js:8]: Why did you put each const into a separate `const` statement instead of a comma separted list? in admin.js above, you did it as a comma separated list (most probably because of copy/paste form the origin)
I like the comma separated list instead (which seems to be done all the other times in the scripts already.
Let's be consistent and also have a comma separated list here 🙂

File: tasks/admin/components/init.js (hunk 7 above)
Comment (mvorisek) [reply tasks/admin/components/init.js:40]: Try moving this require up, local build will fail, as the require tries to connect to GitHub. It is good practise in libs to only define things, but no execute them immediately. Can you rewrite the `config/admin/github.js` to a singleton pattern?

File: tasks/collections/build.js (hunk 11 above)
Comment (mvorisek) [reply tasks/collections/build.js]: I would say single declaration per variable is better for diff, readability, ... But definitely not to be changed here nor now, probably to be changed in #2584. Here fixed.

File: tasks/admin/components/init.js (hunk 7 above)
Comment (lubber-de) [reply tasks/admin/components/init.js:40]: Ok, understood. The whole github API integration seems to be broken anyway and using deprecated functions, so i approve your changes as is for now as it needs a refactoring separately and is not a task for local building but for repo-admin purposes only

**Review by lubber-de — APPROVED** [review]
LGTM
